### PR TITLE
Added inline demo to README

### PR DESCRIPTION
**Pull request by dhrytsenko**
Notes:

- add webcomponent badge and inline demo
- create description about element for documentation (http://neovici.github.io/cosmoz-moment)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,58 @@
...
+<!---
+```
+<custom-element-demo>
+  <template>
+    <script src="../webcomponentsjs/webcomponents-lite.js"></script>
+	<link rel="import" href="../neon-animation/web-animations.html">
+	<link rel="import" href="../paper-dropdown-menu/paper-dropdown-menu.html">
+	<link rel="import" href="../paper-listbox/paper-listbox.html">
+	<link rel="import" href="../paper-item/paper-item.html">
+    <link rel="import" href="cosmoz-moment.html">
```
Comment (nomego) [inline README.md]: Space indent

File: README.md [hunk 2]
```diff
@@ -1,10 +1,58 @@
...
+    <link rel="import" href="cosmoz-moment-timeago.html">
+	<style is="custom-style">
```
Comment (nomego) [inline README.md]: Is this really needed? wc.org encourages responsive demos

File: README.md [hunk 3]
```diff
@@ -1,10 +1,58 @@
...
+		}
+		paper-dropdown-menu, .message {
+			margin-left: 20px;
+		}
+
+		paper-dropdown-menu {
+			height: 80px;
+		}
+	</style>
+    <div id="container">
```
Comment (nomego) [inline README.md]: container shouldn't be needed

File: README.md [hunk 4]
```diff
@@ -1,10 +1,58 @@
...
+      <next-code-block></next-code-block>
+    </div>
+  </template>
+</custom-element-demo>
+```
+-->
+```html
+<template is="dom-bind">
+	<paper-dropdown-menu label="Locale" value="{{ locale }}">
+		<paper-listbox class="dropdown-content" slot="dropdown-content" selected="0">
```
Comment (nomego) [inline README.md]: `slot` is for 2.0, we should only add that when we upgrade (and drop `class`)

File: README.md [hunk 5]
```diff
@@ -17,6 +65,15 @@ cosmoz-moment is a Polymer component for centralized management of Moment.js wit
 <link rel="import" href="bower_components/cosmoz-moment/cosmoz-moment.html" />
 ```
 
+### Use it inside your element
```
Comment (nomego) [inline README.md]: cosmoz-moment should only be used once per _app_ to manage the locale and moment.js.
cosmoz-moment-timeago can be used within elements, or directly in the app, multiple times.
Another (better?) use case is to implement the Cosmoz.MomentBehavior in an element to be able to compute on `locale` and use moment.js-helpers.

File: README.md [hunk 6]
```diff
@@ -17,6 +65,15 @@ cosmoz-moment is a Polymer component for centralized management of Moment.js wit
...
+```html
+<cosmoz-moment locale="[[locale]]"></cosmoz-moment>
+```
+
 ## Docs
```
Comment (nomego) [inline README.md]: Docs will be handled by wc.org, not sure what their guidelines say about this, maybe omit it?

File: cosmoz-moment.html [hunk 7]
```diff
@@ -1,8 +1,27 @@
-<link rel="import" href="../polymer/polymer.html">
 <!--
+@license
+Copyright (c) 2015 Neovici AB. All rights reserved.
```
Comment (nomego) [inline cosmoz-moment.html]: 2017

File: cosmoz-moment.html [hunk 8]
```diff
@@ -12,10 +31,10 @@
 		window.Cosmoz = window.Cosmoz || {};
 
 		var localSharedMoment = moment(),
-			locale = 'en',
```
Comment (nomego) [inline cosmoz-moment.html:15]: indent should not be changed

File: cosmoz-moment.html [hunk 9]
```diff
@@ -65,18 +84,24 @@
 		Polymer({
 			is: 'cosmoz-moment',
 			properties: {
+				/**
+				 * Locale abbreviation for Moment.js locale
+				 *
+				 * @attribute locale
```
Comment (nomego) [inline cosmoz-moment.html]: none of these should be needed

File: cosmoz-moment.html [hunk 10]
```diff
@@ -65,18 +84,24 @@
...
+				 * @type string
+				 * @default 'en'
+				 */
 				locale: {
 					type: String,
+					value: function () {
```
Comment (nomego) [inline cosmoz-moment.html]: no need to run a function for String, if you don't, default value will be detectable, too

File: demo/index.html [hunk 11]
```diff
@@ -32,7 +33,7 @@
 		<demo-snippet>
 			<template is="dom-bind">
 				<paper-dropdown-menu label="Locale" value="{{ locale }}">
-					<paper-listbox class="dropdown-content" selected="0">
+					<paper-listbox class="dropdown-content" slot="dropdown-content" selected="0">
```
Comment (nomego) [inline demo/index.html]: slot for 2.0, leave it for now, will replace class later

File: test/cosmoz-moment.html [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+  <script>
+	suite('cosmoz-moment properties', function () {
+		var baseFixture = fixture('basic');
+
+		test('instantiating the element works', function () {
+			assert.equal(baseFixture.is, 'cosmoz-moment');
+		});
+
+		test('get default locale value', function () {
```
Comment (nomego) [inline test/cosmoz-moment.html]: should probably mention the expected value - "default locale is 'en'" or something

File: test/cosmoz-moment.html [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+			assert.isString(baseFixture.locale);
+			assert.equal(baseFixture.locale, "en");
+		});
+	});
+
+	suite('binding to locale property', function() {
```
Comment (nomego) [inline test/cosmoz-moment.html]: would also be good with a test that sets the locale on cosmoz-moment and then observes a change in a cosmoz-moment-timeago

File: test/cosmoz-moment.html [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+</head>
+
+<body>
+	<test-fixture id="basic">
+		<template>
+			<cosmoz-moment></cosmoz-moment>
+		</template>
+	</test-fixture>
+
+	<template is="dom-bind" id="custom-locale">
```
Comment (nomego) [inline test/cosmoz-moment.html]: shouldn't this also be in a test-fixture?

**Review by nomego — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (dhrytsenko) [reply README.md]: l added it for extra space on bottom cause dropdown list will have scrollbar in webcomponents preview. it's not critical and l will remove it

File: README.md (hunk 4 above)
Comment (dhrytsenko) [reply README.md]: Yes, it's make sence

File: README.md (hunk 6 above)
Comment (dhrytsenko) [reply README.md]: l think we can leave it if we have plans to add more complex examples. This link to docs will show `gh-pages` with property list and some additional examples. `gh-pages` provide live docs and previews of our element via GitHub pages.

File: test/cosmoz-moment.html (hunk 13 above)
Comment (dhrytsenko) [reply test/cosmoz-moment.html]: yes, it'll be nice. l added basic tests for cosmoz-moment-timeago and will try to create what you describe

File: test/cosmoz-moment-timeago.html [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+		});
+	});
+
+	suite('set date property', function() {
+		var customDate = fixture('custom-date');
+		test('should show time ago from date', function () {
+			var timeagoSpan = customDate.$$('span');
+			assert.isNotNull(timeagoSpan);
+			assert.isString(timeagoSpan.textContent);
+            assert.notEqual(timeagoSpan.textContent, 'Invalid date', 'date has invalid value');
```
Comment (nomego) [inline test/cosmoz-moment-timeago.html]: spacing indent

File: test/index.html [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+	<title>cosmoz-moment tests</title>
+	<script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+	<script src="../../web-component-tester/browser.js"></script>
+</head>
+<body>
+	<script>
+	/*global WCT*/
+		WCT.loadSuites([
+			'cosmoz-moment.html',
+            'cosmoz-moment-timeago.html',
```
Comment (nomego) [inline test/index.html]: spacing indent

**Review by nomego — CHANGES_REQUESTED** [review]
All comments from the last review are not addressed still

**Comment by nomego** [comment]
Also, build is failing.

**Comment by dhrytsenko** [comment]
There are some issues in Firefox while run tests. l work on it

**Comment by dhrytsenko** [comment]
@nomego, I fixed travis build issue. Could you check it again, please?

File: cosmoz-moment.html (hunk 8 above)
Comment (nomego) [reply cosmoz-moment.html:15]: Not addressed

File: cosmoz-moment.html [hunk 17]
```diff
@@ -12,10 +31,10 @@
 		window.Cosmoz = window.Cosmoz || {};
 
 		var localSharedMoment = moment(),
```
Comment (nomego) [inline cosmoz-moment.html:33]: Does not seem used anymore? Drop in that case

File: demo/index.html (hunk 11 above)
Comment (nomego) [reply demo/index.html]: Not addressed (do not add the `slot` attribute)

**Review by nomego — CHANGES_REQUESTED** [review]

File: cosmoz-moment.html (hunk 17 above)
Comment (dhrytsenko) [reply cosmoz-moment.html:33]: We can't remove it cause use it in `MomentBehavior` to set local moment variable:
 
    `Cosmoz.MomentBehavior = {
         properties: {
		locale: {
			type: String,
			readOnly: true,
			value: function () {
				return locale;
			}
		},
		moment: {
			type: Object,
			value: localSharedMoment
		}
	},`

File: cosmoz-moment.html (hunk 8 above)
Comment (dhrytsenko) [reply cosmoz-moment.html:15]: Is it correct indent: 

    `var localSharedMoment = moment(),
     locale = 'en',
     momentElements = [];` 

or you want this:

    `var localSharedMoment = moment(),
		 locale = 'en',
		 momentElements = [];` 

Now we have first indent for variables

File: cosmoz-moment.html (hunk 17 above)
Comment (nomego) [reply cosmoz-moment.html:33]: In "Added component and variable description for docs"(bfd23ec8f7b8eaca646ccbb75c920549c66b65a9) that property is removed.

File: cosmoz-moment.html (hunk 8 above)
Comment (nomego) [reply cosmoz-moment.html:15]: The last one. It was correct before the PR, see https://github.com/Neovici/cosmoz-moment/pull/2/files

File: cosmoz-moment.html (hunk 17 above)
Comment (dhrytsenko) [reply cosmoz-moment.html:33]: it was removed from cosmoz-moment component properties but still exist in behavior properties

Comment (nomego) [reply cosmoz-moment.html:33]: Oh ok, I see.

**Review by nomego — APPROVED** [review]

**Pull request merged**